Automattic/wp-calypso
# State: Move gravatar status reducer out of current user reducer

#### Changes proposed in this Pull Request

Currently, much of Calypso's reducers are modular and are loaded on demand when needed. The only bits that form the monolithic reducer nowadays are:

* Current user
* Data layer
* Sites

As part of the monolithic reducer, these are always loaded on every Calypso page load.

The gravatar status reducer is part of the current user reducer, but it's not connected to it in any way. It's also used only in a couple of places - when displaying an avatar and when uploading a new one. 

So this PR splits out the gravatar status reducer out of the current user one. It also changes the gravatar status to be a modular one.

#### Testing instructions

* Verify Calypso builds well.
* Go to `/me`
* Verify you can still upload an avatar like before.
* Verify all tests pass.
